lib/cors-anywhere.js

```diff
@@ -59,8 +59,10 @@ if (providerJSONFileExists) {
   });
 }
 
-const isProviderAllowed = (provideEndpoint) => {
-  return !!whitelistEndpoints[provideEndpoint];
+const isProviderAllowed = (providerEndpoint) => {
+  providerEndpoint = providerEndpoint.replace('.well-known/smart-configuration', '');
+  providerEndpoint = providerEndpoint.replace('metadata', '');
+  return !!whitelistEndpoints[providerEndpoint];
```

Comment:
This might work for our limited use if we find all the ways we use the location and add similar code, but really we ought to do a generalized algorithm, so we don't have to limit use cases to known paths only.

If we sort the list on load and binary search the list to find where the providerEndpoint fits we can determine a match.  We can cache matches for quick O(1) lookup but the initial binary search really isn't too expensive.  If the incoming list is sorted at load time the search time will be O(logN).  NOTE: for our list of 7800 Log is about 9, so it takes at most 9 lookups to find the item.  It should still be pretty darn fast.

Here's a start to a proper algorithm without caching.  https://chatgpt.com/share/e/6733f2db-89d8-8005-911a-d1037b42ef7a